great_expectations/core/expectation_suite.py

```diff
@@ -144,27 +144,42 @@ def add_expectation(self, expectation: _TExpectation) -> _TExpectation:
                 "and set `Expectation.id = None`."
             )
         should_save_expectation = self._has_been_saved()
-        expectation_is_unique = all(
-            expectation.configuration != existing_expectation.configuration
-            for existing_expectation in self.expectations
-        )
-        if expectation_is_unique:
+
+        already_added, _ = self._contains_equalish_expectation(expectation)
+        if not already_added:
             # suite is a set-like collection, so don't add if it not unique
-            self.expectations.append(expectation)
             if should_save_expectation:
-                try:
-                    expectation = self._store.add_expectation(suite=self, expectation=expectation)
-                    self.expectations[-1].id = expectation.id
-                except Exception as exc:
-                    self.expectations.pop()
-                    raise exc  # noqa: TRY201
+                expectation = self._store.add_expectation(suite=self, expectation=expectation)
+            self.expectations.append(expectation)
 
         expectation.register_save_callback(save_callback=self._save_expectation)
 
         self._submit_expectation_created_event(expectation=expectation)
 
         return expectation
 
+    def _contains_equalish_expectation(self, expectation: Expectation) -> tuple[bool, list[int]]:
+        """
+        Helper method to determine if an expectation is already in the suite.
+
+        Note that this check is less stringent than Expectation.__eq__ and excludes
+        a few fields that are not relevant for uniqueness in the suite.
+
+        Args:
+            expectation: The expectation to check for.
+
+        Returns:
+            A tuple of a boolean indicating whether the expectation is already in the suite
```

Comment:
Non-blocking: the method return type feels like a bit of a smell, and designed for a very specific use case. Would it be simpler if we took in 2 expectations, and just returned a bool, then let `delete_expectation` be responsible for looping?

Don't feel like you need to over-index on this - we've already had a ton of back and forth, and I appreciate your patience!!